app/src/main/scala/org/alephium/explorer/persistence/schema/OutputSchema.scala

```diff
@@ -60,9 +63,13 @@ object OutputSchema extends SchemaMainChain[OutputEntity]("outputs") {
        mainChain,
        lockTime,
        outputOrder,
-       txOrder)
+       txOrder,
+       spentFinalized)
         .<>((OutputEntity.apply _).tupled, OutputEntity.unapply)
   }
 
+  lazy val createNonSpentIndex: DBActionW[Int] =
+    sqlu"create index if not exists non_spent_output_idx on #${name} (main_chain,address,key, block_hash) where spent_finalized IS NULL;"
```

Comment:
Nit: one extra space inside `main_chain,address,key, block_hash`

Reply:
by adding `block_hash`, I am afraid that the index will not be used for the address query since `block_hash` is not used . @simerplaha we will need your help to investigate this part.

Reply:
Quote from Thomas's comment:
```
This new index is still used in the `getBalance` query, confirmed with an `EXPLAIN` command
```

But let's double check this as I am not sure how the query could use the index when `block_hash` is not used

Reply:
I found the answer from the doc of Postgresql. Multi-column indexes can be used for queries that involve a subset of the columns, but the order of the columns matters here.
Quote from Postgresql: https://www.postgresql.org/docs/current/indexes-multicolumn.html
```
A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, but the index is most efficient when there are constraints on the leading (leftmost) columns.
```

We should probably make `address` the first column for this index? cc @tdroxler @simerplaha

Reply:
but I am not sure if the index is a B-tree index or other type of index

Reply:
![maim-1650543338](https://user-images.githubusercontent.com/2979182/164456189-933ec58d-0a0f-4d46-a412-d8a61bbdf1ae.png)

on the image, the first explain is with the current index, I then remove it and create with `address` first. The next explain is then similar. The first `Index Cond` is just swapped, don't know if it changes something?

Reply:
According to the doc, the order of the index conditions matters for indexes like B-tree. Could you check if the index type is `btree`? If so, then we'd better put `address` as the first column of the index.

Ref: https://www.postgresqltutorial.com/postgresql-indexes/postgresql-list-indexes/

Reply:
![maim-1650560351](https://user-images.githubusercontent.com/2979182/164512686-2bb71665-2280-4441-b55e-c236dc0396fc.png)
 yep it's a b tree, I'll update the index.